web/impact/impact/views/algolia_api_key_view.py

```diff
@@ -59,14 +59,19 @@ def get(self, request, format=None):
             'validUntil': int(time.time()) + 3600,
             'userToken': request.user.id,
         }
+        programs_finalist_in = _get_finalist_program_role_grants(
+            request.user).values_list('program_role__program__name', flat=True)
         if filters:
             params['filters'] = filters
         public_key = _get_public_key(params, search_key)
-        return Response({
+        response_data = {
             'token': public_key,
             'index_prefix': settings.ALGOLIA_INDEX_PREFIX,
-            'filters': filters
-        })
+            'filters': filters,
+        }
+        if programs_finalist_in.exists():
```

Comment:
Is this check necessary, why do we care whether the list contains something or not?